spark-3.1-spanner-lib/src/test/java/com/google/cloud/spark/spanner/ReadIntegrationTestBase.java

```diff
@@ -90,6 +100,77 @@ public void testReadFromCompositeTable() {
     assertThat(gotDs).containsExactlyElementsIn(expectDs);
   }
 
+  @Test
+  public void testDatasetFilters_schema() {
+    Dataset<Row> df = readFromTable("compositeTable");
+    StructType gotSchema = df.schema();
+    StructType expectSchema =
+        new StructType(
+            Arrays.asList(
+                    new StructField("id", DataTypes.StringType, false, null),
+                    new StructField(
+                        "A", DataTypes.createArrayType(DataTypes.LongType, true), true, null),
+                    new StructField(
+                        "B", DataTypes.createArrayType(DataTypes.StringType, true), true, null),
+                    new StructField("C", DataTypes.StringType, true, null),
+                    new StructField("D", DataTypes.createDecimalType(38, 9), true, null),
+                    new StructField("E", DataTypes.DateType, true, null),
+                    new StructField("F", DataTypes.TimestampType, true, null),
+                    new StructField("G", DataTypes.BooleanType, true, null),
+                    new StructField(
+                        "H", DataTypes.createArrayType(DataTypes.DateType, true), true, null),
+                    new StructField(
+                        "I", DataTypes.createArrayType(DataTypes.TimestampType, true), true, null))
+                .toArray(new StructField[0]));
+
+    // For easy debugging, let's firstly compare the .treeString() values.
+    // Object.equals fails for StructType with fields so firstly
+    // compare lengths, then fieldNames then the treeString.
+    assertThat(gotSchema.length()).isEqualTo(expectSchema.length());
+    assertThat(gotSchema.fieldNames()).isEqualTo(expectSchema.fieldNames());
+    assertThat(gotSchema.treeString()).isEqualTo(expectSchema.treeString());
+  }
+
+  @Test
+  public void testDatasetFilters_sort() {
```

Comment:
I'd call it column select. It's not filters. Filters are the condition inside where() or filter() function.

Reply:
Addressed and changed, thank you @halio-g!